src/generate/impl.rs

```diff
@@ -83,9 +83,9 @@ impl<'a> Impl<'a, Generator> {
     {
         if let Some(generics) = self.parent.generics() {
             let mut constraints = self
-                .parent
-                .generic_constraints()
-                .cloned()
+                .custom_generic_constraints
+                .take()
+                .or_else(|| self.parent.generic_constraints().cloned())
```

Comment:
I think this can be
```rs
let constraints = self.custom_generic_constraints.get_or_insert_with(|| self.parent.generic_constraints().cloned().unwrap_or_default());
cb(generics, constraints);
// drop the `self.custom_generic_constraints = Some(constraints);
```